src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -424,3753 +416,3744 @@ private void BuildApplicationInitializerBody(IndentedStringBuilder writer, XamlO
+			writer.AppendLineInvariant("// WARNING " + message, options);
+		}
+
+		private bool HasMarkupExtension(XamlMemberDefinition valueNode)
+		{
+			// Return false if the Owner is a custom markup extension
+			if (IsCustomMarkupExtensionType(valueNode.Owner?.Type))
+			{
+				return false;
+			}
+
+			return valueNode
+				.Objects
+				.Any(o =>
+					o.Type.Name == "StaticResource"
+					|| o.Type.Name == "ThemeResource"
+					|| o.Type.Name == "Binding"
+					|| o.Type.Name == "Bind"
+					|| o.Type.Name == "TemplateBinding"
+				);
+		}
+
+		private bool HasCustomMarkupExtension(XamlMemberDefinition valueNode)
+		{
+			// Verify if a custom markup extension exists
+			return valueNode
+				.Objects
+				.Any(o => IsCustomMarkupExtensionType(o.Type));
+		}
+
+		private bool HasBindingMarkupExtension(XamlMemberDefinition valueNode)
+		{
+			return valueNode
+				.Objects
+				.Any(o =>
+					o.Type.Name == "Binding"
+					|| o.Type.Name == "Bind"
+					|| o.Type.Name == "TemplateBinding"
+				);
+		}
+
+		private bool IsCustomMarkupExtensionType(XamlType xamlType)
+		{
+			if (xamlType == null)
+			{
+				return false;
+			}
+
+			// Adjustment for Uno.Xaml parser which returns the namespace in the name
+			var xamlTypeName = xamlType.Name.Contains(':') ? xamlType.Name.Split(':').LastOrDefault() : xamlType.Name;
+
+			// Determine if the type is a custom markup extension
+			return _markupExtensionTypes.Any(ns => ns.Name.Equals(xamlTypeName, StringComparison.InvariantCulture));
+		}
+
+		private XamlMemberDefinition FindMember(XamlObjectDefinition xamlObjectDefinition, string memberName)
+		{
+			return xamlObjectDefinition.Members.FirstOrDefault(m => m.Member.Name == memberName);
+		}
+
+		private XamlMemberDefinition GetMember(XamlObjectDefinition xamlObjectDefinition, string memberName)
+		{
+			var member = FindMember(xamlObjectDefinition, memberName);
+
+			if (member == null)
+			{
+				throw new InvalidOperationException($"Unable to find {memberName} on {xamlObjectDefinition}");
+			}
+
+			return member;
+		}
+
+		private (string ns, string className) GetClassName(XamlObjectDefinition control)
+		{
+			var classMember = FindClassName(control);
+
+			if (classMember.ns == null)
+			{
+				throw new Exception("Unable to find class name for toplevel control");
+			}
+
+			return classMember;
+		}
+
+		private static (string ns, string className) FindClassName(XamlObjectDefinition control)
+		{
+			var classMember = control.Members.FirstOrDefault(m => m.Member.Name == "Class");
+
+			if (classMember != null)
+			{
+				var fullName = classMember.Value.ToString();
+
+				var index = fullName.LastIndexOf('.');
+
+				return (fullName.Substring(0, index), fullName.Substring(index + 1));
+			}
+			else
+			{
+				return (null, null);
+			}
+		}
+
+		private bool BuildProperties(IIndentedStringBuilder writer, XamlObjectDefinition topLevelControl, bool isInline = true, bool returnsContent = false, string closureName = null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change return type to 'void'; not a single caller uses the returned value.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5305744)